documentation/general/dash-sonic-hld.md

```diff
@@ -1527,3 +1575,61 @@ For the example configuration above, the following is a brief explanation of loo
+2. Enable Continuous Integration(CI) via Azure Pipelines (Azp) for SONiC repositories, like sonic-buildimage, sonic-swss and so on.
+
+## 4.2 Architecture
+
+![BMv2 virtual SONiC](images/bmv2-virtual-sonic.svg)
+
+## 4.3 Modules
+
+### 4.3.1 BMv2 (dataplane engine)
+
+This component is the original P4 BMv2 container image, which serves as the implementation of the dataplane.
+It attaches three types of interfaces: Ethernet, eth, and cpu.
+- Ethernet is used as the system port. Protocol services like BGP and LLDP perform send/receive operations on these interfaces.
+- eth is used as the line port. These are native interfaces in KVM for communication between the inside and outside. The eth port and Ethernet port is one-to-one mapping.
+- cpu is used for the dpdk port. The dataplane APP directly manipulates the traffic on these ports.
+
+### 4.3.2 Dataplane APP
+
+In the physical DPU, this component is provided by vendors' DPDK application. However, in the virtual switch, our implementation is based on the VPP framework with the CPU interface to simulate it. This dataplane APP loads the generated shared library, saidash.so, which communicates with BMv2 via GRPC. Additionally, this component connects with sairedis through a shim SAI proxy.
+
+### 4.3.3 SAIRedis
+
+In the original virtual SONiC, SAIRedis will load the saivs.so. However, in the SONiC DASH virtual switch, it will load the sai proxy mentioned above.
+
+### 4.3.4 SWSS
+
+The SWSS on the virtual switch is almost the same as the one used in the physical switch (DPU). We don't need to make any special changes to it. However, due to some tedious historic issues, the SWSS on the virtual switch will make some extra SAI calls (like DTEL). To adapt to these calls, we will generate a corresponding mock SAI implementation in saidash.so.
+
+### 4.3.5 Underlay Service
+
+We plan to keep the underlay services, such as BGP, LLDP, and others. This is because we not only want to simulate a DPU but also provide an opportunity to use a virtual SONiC image to simulate a smart switch with a single DPU for simplified scenarios.
+
+## 4.4 Dataflow
+### 4.4.1 Data plane
+
+All data plane traffic will enter the BMv2 simple switch and be forwarded to the target port based on the P4 logic imported on BMv2.
+
+```text
+E.G.
+
+LLDP packet:
+  eth1<->Ethernet0<->LLDP process
+
+VNet packet:
+  eth1<->dash pipeline(BMv2)<->eth2
+
+TCP SYN packet:
+  eth1<->cpu0<->dataplane APP<->cpu0<->eth2
+```
+
+### 4.4.2 Control plane
+
+In the physical SmartSwitch, configuration is forwarded via GNMI in the NPU. So, in the virtual SONiC environment, the SWSS module is capable of receiving configuration from an external GNMI service through the management port, eth-midplane. However, in the single device mode, the GNMI service can also be run within the KVM and directly forward the configuration to the local SWSS module.
```

Comment:
the gNMI is actually missing in the architecture graph, do you mind to add it there? I think it is way more important than BGP, since we are not using BGP on DPU anyway in SmartSwitch setup.

Reply:
Done

Reply:
Done